app/templates/_exchanges.html

```diff
@@ -0,0 +1,290 @@
+                            <button
+                                onclick="deleteCheck('{{ ex.id }}', 'cancel', -1, -1, '{{ user.hospital.id }}') "
+                                type="button" class="btn btn-danger btn-sm">Cancel</button>
+                        {% endcall %}
+                    {% else %}
+                        {{ exchange_status_and_actions(ex, e) }}
+                    {% endif %}
+                {% endif %}
+            </tr>
+            {% endfor %}
+            {% endfor %}
+        </tbody>
+    </table>
+{% endmacro %}
+
+{% block exchanges %}
+<div class="well current-exchanges">
+    <h2>Your Exchanges</h2>
+    {{ exchanges_listing('Actionable', 'Exchanges will appear here when they require your input to move forward.', exchanges['actionable']) }}
+    {{ exchanges_listing('Pending', "Exchanges will appear here when they're in progress, but don't require anything from you right now.", exchanges['pending']) }}
+    {{ exchanges_listing('Completed', "Exchanges will appear here when they're completely finished.", exchanges['complete']) }}
+</div>
+
+
+<div id="delete-check" class="modal" tabindex="-1" role="dialog">
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h2 class="modal-title">Cancel Confirmation</h2>
+            </div>
+            <div class="modal-body">
+                <p>Are you sure you want to cancel Exchange <span id="remove-item-name"></span></p>
+            </div>
+            <div class="modal-footer">
+                <button id="yes" class="btn btn-danger" onclick="updateExchange2()">Yes</button>
+                <button id="no" class="btn btn-primary" onclick="hideDeleteCheck()">No</button>
+            </div>
+        </div>
+    </div>
+</div>
+
+<script>
+    function fatalError() {
+        alert("There was an error. Your changes may not have been saved.\n\nReloading...");
+        window.document.location.reload();
+    }
+    var data = {};
+    $("#delete-check").hide()
+
+    function deleteCheck(exchange_id, task, e_id, number, hospital_id) {
+        data = { exchange_id: exchange_id, task: task, e_id: e_id, number: number, hospital_id: hospital_id, user_id: '{{ user.id }}' };
+        $("#delete-check").show();
+        $("#remove-item-name").text(exchange_id);
+    }
+
+    function updateExchange2() {
+        $.ajax({
+            url: "/update_exchange",
+            type: "POST",
+            contentType: 'application/json',
+            data: JSON.stringify(data),
+            dataType: 'json',
+            success: function (result) {
+                location.reload();
+            },
+            error: fatalError
+        });
+    }
+
+
+    function updateExchange(exchange_id, task, e_id, number, hospital_id) {
+        data = { exchange_id: exchange_id, task: task, e_id: e_id, number: number, hospital_id: hospital_id , user_id: '{{ user.id }}'};
```

Comment:
Although I didn't want to go through and rip it out, best I can tell, the `number` parameter isn't ever used anywhere.